### DISCUSSION TITLE
Page changes faster than Layout

### DISCUSSION THREAD

**Discussion by ilyaDegtyarenko** [nuxt/nuxt#16769] [Questions]
This is a problem because the content in the layouts is different
Why is this happening?

Reproduced in the example:
[https://stackblitz.com/edit/nuxt-starter-yvx4ax?file=app.vue](https://stackblitz.com/edit/nuxt-starter-yvx4ax?file=app.vue)

**Comment by sevillaarvin** [comment]
I believe it's because of a typo (the bottom **layout-leave-active** should be **layout-leave**):

before
```
.layout-enter-active,
.layout-leave-active {
  transition: opacity 1s;
}

.layout-enter,
.layout-leave-active {
  opacity: 0;
}
```

after
```
.layout-enter-active,
.layout-leave-active {
  transition: opacity 1s;
}

.layout-enter,
.layout-leave {
  opacity: 0;
}
```

**Reply by sevillaarvin** [reply]
EDIT: It has to be the following to animate correctly:

```
.layout-enter-from,
.layout-leave-to {
  opacity: 0;
}
```

References:

https://vuejs.org/guide/built-ins/transition.html#the-transition-component

**Reply by ilyaDegtyarenko** [reply]
> I believe it's because of a typo (the bottom **layout-leave-active** should be **layout-leave**):
> 
> before
> 
> ```
> .layout-enter-active,
> .layout-leave-active {
>   transition: opacity 1s;
> }
> 
> .layout-enter,
> .layout-leave-active {
>   opacity: 0;
> }
> ```
> 
> after
> 
> ```
> .layout-enter-active,
> .layout-leave-active {
>   transition: opacity 1s;
> }
> 
> .layout-enter,
> .layout-leave {
>   opacity: 0;
> }
> ```

**Changed to:**
```
.layout-enter-active,
.layout-leave-active {
  transition: opacity 1s;
}

.layout-enter-to {
  opacity: 1;
}

.layout-leave-to {
  opacity: 0;
}
```
Look please [https://stackblitz.com/edit/nuxt-starter-yvx4ax](https://stackblitz.com/edit/nuxt-starter-yvx4ax?file=app.vue)

Same behavior

**Reply by ilyaDegtyarenko** [reply]
> EDIT: It has to be the following to animate correctly:
> 
> ```
> .layout-enter-from,
> .layout-leave-to {
>   opacity: 0;
> }
> ```
> 
> References:
> 
> https://vuejs.org/guide/built-ins/transition.html#the-transition-component

Look at the changes please [https://stackblitz.com/edit/nuxt-starter-yvx4ax](https://stackblitz.com/edit/nuxt-starter-yvx4ax?file=app.vue)

Same behavior

**Reply by sevillaarvin** [reply]
How about this? https://stackblitz.com/edit/nuxt-starter-n8mquq?file=app.vue

Not sure if this is the best way. Also take a look at https://github.com/nuxt/framework/discussions/851 see if it helps.
